sway/config.c

```diff
@@ -300,6 +326,9 @@ static void config_defaults(struct sway_config *config) {
 	if (!(config->feature_policies = create_list())) goto cleanup;
 	if (!(config->ipc_policies = create_list())) goto cleanup;
 
+	// The keysym to keycode translation
+	config->keysym_translation = new_keysym_translation_data(getenv("XKB_DEFAULT_LAYOUT"));
```

Comment:
The `XKB_DEFAULT_LAYOUT` env variable is deprecated. Also, what about xkb options?

Can't we just reference use the first xkb layout we create from the config instead? Why create another one?

Reply:
Yep, I will need to add xkb options.

As for the reference, ideally we need to use `xkb_keymap` from the corresponding `sway_keyboard` instance. The issue is that input devices are created after the main config is loaded. To workaround this, we might need to add some code to update config and retranslate keysyms when the keyboard is configured somewhere in `seat_add_device`, but in my opinion a separate instance of `xkb_keymap` is not so expensive to justify the complications.